src/tiledb/cloud/bioimg/exportation.py

```diff
@@ -127,41 +169,45 @@ def export_tiff_udf(
                 **kwargs,
             )
 
-    if isinstance(source, str):
-        # Handle only lists
-        source = [source]
-    logger.debug("Exporting files: %s", source)
+    source = [source] if isinstance(source, str) else source
+    output = [output] if isinstance(output, str) else output
+    validate_io_paths(source, output, for_registration=False)
 
     # Build the task graph
     dag_name = taskgraph_name or DEFAULT_DAG_NAME
 
     logger.debug("Building graph")
 
-    run_mode = run_on or "server"
-    if run_mode not in _RUNNING_PROFILES:
-        raise ValueError("Invalid value for argument 'run_on'")
-
     graph = dag.DAG(
         name=dag_name,
-        mode=dag.Mode.REALTIME if run_mode == "client" else dag.Mode.BATCH,
+        mode=mode,
         max_workers=max_workers,
         namespace=namespace,
+        retry_strategy=RetryStrategy(
+            limit=3,
+            retry_policy="Always",
+        ),
     )
 
+    if not access_credentials_name:
+        raise ValueError(
+            "Ingestion graph requires `access_credentials_name` to be set."
+        )
+
```

Comment:
@ktsitsi Is this the soonest we can validate the argument? Let's move it as high in the function as we can so that we can fail as quickly as possible.

Reply:
Addressed!